File: webview/zed/phase0/scratch/Sample.java
package phase0;

/**
 * Phase 0 scratch file: the target of the stub language server's window/showDocument and
 * workspace/applyEdit requests. Disposable — nothing in the build reads it.
 */
public class Sample {

    public static void main(String[] args) {
        System.out.println("webview phase 0: the stub asked Zed to put the caret on the next line");
        System.out.println("webview phase 0: the stub then asked Zed to apply an edit above this class");
    }
}

File: webview/zed/phase0/scratch/Sample.rs
// Phase 0 scratch file for the rust-analyzer adapter override.
//
// The repo's .zed/settings.json points lsp.rust-analyzer.binary.path at node + stub-lsp.mjs, so
// opening this file makes Zed start the stub instead of rust-analyzer. The stub then asks Zed for
// window/showDocument (caret to the marked line) and applies one workspace/applyEdit.

fn main() {
    // CARET TARGET: the stub asks for line 10, column 5 of this file.
    println!("webview phase 0");
}
